Assets/Scripts/PlayerController.cs

```diff
@@ -228,6 +237,17 @@ public class PlayerController : MonoBehaviour {
     }
 
     // Helper Methods
+    private void ResetPlayerPosition() {
+        rb.velocity = Vector3.zero;
+        rb.angularVelocity = Vector3.zero;
+        if (this.playerLastPlatform != null) {
+            // If player jumped off a platform but didn't hit any other ground, it should reset to the platform
+            var platformPosition = this.playerLastPlatform.transform.position;
+            playerLastPosition = platformPosition + new Vector3(0, (this.playerLastPlatform.transform.lossyScale.y / 2), 0);
```

Comment:
I was thinking about that, hence the reason why I was storing the players "last object" and potentially storing the offset, but it turned out to be pretty hard (but doable)